plugins/inputs/slurm/slurm_test.go

```diff
@@ -0,0 +1,399 @@
+		plugin := Slurm{
+			URL: url,
+		}
+		require.NoError(t, plugin.Init())
+	}
+
+	for _, url := range []string{
+		"httpp://example.com:6820", "httpss://example.com:6820", "", "http://:6820", "http://",
+	} {
+		plugin := Slurm{
+			URL: url,
+		}
+		require.Error(t, plugin.Init())
+	}
+}
+
+func TestPanicHandling(t *testing.T) {
+	ts := httptest.NewServer(http.NotFoundHandler())
+	defer ts.Close()
+
+	ts.Config.Handler = http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		switch r.URL.Path {
+		case "/slurm/v0.0.38/diag":
+			w.Header().Add("Content-Type", "application/json")
+			w.WriteHeader(http.StatusOK)
+			_, err := w.Write([]byte(`{
+				"meta": {},
+				"errors": [],
+				"statistics": {
+					"rpcs_by_message_type": [],
+					"rpcs_by_user": [],
+					"jobs_running": 100
+				}
+			}`))
+			require.NoError(t, err)
+		default:
+			w.WriteHeader(http.StatusNotFound)
+			t.Fatalf("unexpected path: " + r.URL.Path)
+		}
+	})
+
+	plugin := &Slurm{
+		URL:              "http://" + ts.Listener.Addr().String(),
+		IgnoredEndpoints: []string{"jobs", "nodes", "partitions", "reservations"},
+	}
+	require.NoError(t, plugin.Init())
+
+	var acc testutil.Accumulator
+	require.NotPanics(t, func() {
+		require.NoError(t, plugin.Gather(&acc))
+	})
+}
+
+func TestGatherDiagMetrics(t *testing.T) {
+	var (
+		bfActive                = false
+		bfQueueLen        int32 = 1
+		bfQueueLenMean    int32 = 1
+		jobsCanceled      int32 = 3
+		jobsCompleted     int32 = 396
+		jobsFailed        int32 = 2
+		jobsPending       int32 = 10
+		jobsRunning       int32 = 100
+		jobsStarted       int32 = 396
+		jobsSubmitted     int32 = 396
+		scheduleCycleLast int32 = 301
+		scheduleCycleMean int32 = 137
+		serverThreadCount int32 = 3
+	)
+	diag := goslurm.V0038DiagStatistics{
+		BfActive:          &bfActive,
+		BfQueueLen:        &bfQueueLen,
+		BfQueueLenMean:    &bfQueueLenMean,
+		JobsCanceled:      &jobsCanceled,
+		JobsCompleted:     &jobsCompleted,
+		JobsFailed:        &jobsFailed,
+		JobsPending:       &jobsPending,
+		JobsRunning:       &jobsRunning,
+		JobsStarted:       &jobsStarted,
+		JobsSubmitted:     &jobsSubmitted,
+		ScheduleCycleLast: &scheduleCycleLast,
+		ScheduleCycleMean: &scheduleCycleMean,
+		ServerThreadCount: &serverThreadCount,
+	}
+
+	records := make(map[string]interface{})
+	tags := make(map[string]string)
+
+	tags["source"] = "127.0.0.1"
+
+	records["bf_active"] = bfActive
+	records["bf_queue_len"] = bfQueueLen
+	records["bf_queue_len_mean"] = bfQueueLenMean
+	records["jobs_canceled"] = jobsCanceled
+	records["jobs_submitted"] = jobsSubmitted
+	records["jobs_started"] = jobsStarted
+	records["jobs_completed"] = jobsCompleted
+	records["jobs_failed"] = jobsFailed
+	records["jobs_pending"] = jobsPending
+	records["jobs_running"] = jobsRunning
+	records["schedule_cycle_last"] = scheduleCycleLast
+	records["schedule_cycle_mean"] = scheduleCycleMean
+	records["server_thread_count"] = serverThreadCount
```

Comment:
Please implicitly declare the `records` and `tags` maps...

Reply:
Hi! I reworked the entire testing infrastructure following the example you provided. These changes have been added on 24e01e45ccb3ea3fec9dad6a733ca0e678d6a461, so the above no longer applies really...